File: weak2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string input);
int count_words(string input);
int count_sentences(string input);
int index(int sentences,int words,int letters);

int main(void)
{
    printf("Text: ");
    string text=get_string("");
    int letters=count_letters(text);
    int words=count_words(text);
    int sentences=count_sentences(text);
    int index1=index(sentences,words,letters);

    if(index1<=16 && index1>=1)
    {
        printf("Grade %i\n",index1);
    }
    else if(index1 <1)
    {
        printf("Before Grade 1\n");
    }
    else if(index1>16)
    {
        printf("Grade 16+\n");
    }
}


//A function that counts letters contained in a given string
int count_letters(string input)
{
    int count =0;
    int len=strlen(input);
    for(int i=0;i<len;i++)
    {
        if(isalpha(input[i]))
        {
            count++;
        }
    }
    return count;
}


//A function that counts the numbers of words for a given string
int count_words(string input)
{
    int count=1;
    int len=strlen(input);
    for(int i=0;i<len;i++)
    {
        if(isblank(input[i]))
        {
            count+=1;
        }
    }
    return count;
}

//A function that counts the number of  sentances
int count_sentences(string input)
{
    int count=0;
    int len=strlen(input);
    for(int i=0;i<len;i++)
    {
        if(input[i]=='?'||input[i]=='!'||input[i]=='.')
        {
            count++;
        }
    }
    return count;
}

//calculates index = 0.0588 * L - 0.296 * S - 15.8
int index(int sentences,int words,int letters)
{
       return (int)round( (0.0588*(letters/(float)words)* 100)-(0.296*(sentences/(float)words)*100)-15.8);

}

File: weak6/sentimental-readability/readability.py
from cs50 import get_string

def main():
    #get input
    input = get_string("Text: ")
    #calculate letters  , words and sentences
    l = letters(input)
    w = words (input)
    s = sentences (input)

    #calculate the average number of letters for 100 words
    L = (l /(w)) * 100
    #calculate the average number of sentences for 100 words
    S = (s / (w))* 100
    #calculate index : index = 0.0588 * L - 0.296 * S - 15.8
    index = round(0.0588 * L - 0.296 * S - 15.8)
    #print index
    if index < 1:
        print("Before Grade 1")
    if index >= 1 and index <= 16:
        print(f"Grade {index}")
    if index >= 16:
        print ("Grade 16+")




#function that count letters
def letters(string):
    count = 0
    for char in string:
        if (char.isalpha()):
            count += 1
    return count

#function that count words
def words (string):
    count = 1
    for word in string :
        if " " in word:
            count +=1
    return count


#function that count sentences
def sentences(string):
    count=0
    for word in string:
        if "!" in word or "?" in word or "." in word:
            count += 1
    return count


if __name__ == "__main__":
    main()